include/gridtools/common/hymap.hpp

```diff
@@ -194,19 +195,50 @@ namespace gridtools {
     using has_key = meta::st_contains<hymap_impl_::get_keys<Map>, Key>;
 
     namespace hymap {
+        template <class...>
+        struct keys : private enable_maker {
+            template <class...>
+            struct values;
+
+#if !defined(__NVCC__) && defined(__clang__) && __clang_major__ <= 13
+            template <class... Vs>
+            values(Vs const &...) -> values<Vs...>;
+#endif
+
+            static constexpr maker<values> make_values = {};
+        };
+
         template <class... Keys>
-        struct keys {
-            template <class... Vals>
-            struct values {
-                static_assert(sizeof...(Vals) == sizeof...(Keys), "invalid hymap");
+        template <class... Vals>
+        struct keys<Keys...>::values {
+            static_assert(sizeof...(Vals) == sizeof...(Keys), "invalid hymap");
 
-                tuple<Vals...> m_vals;
+            tuple<Vals...> m_vals;
 
-                GT_TUPLE_UTIL_FORWARD_CTORS_TO_MEMBER(values, m_vals);
-                GT_TUPLE_UTIL_FORWARD_GETTER_TO_MEMBER(values, m_vals);
+            template <class... Args,
+                std::enable_if_t<std::conjunction_v<std::is_constructible<Vals, Args>...>, int> = 0>
+            constexpr GT_FUNCTION values(Args &&...args) noexcept : m_vals{wstd::forward<Args>(args)...} {}
 
-                friend keys hymap_get_keys(values const &) { return {}; }
-            };
+            constexpr GT_FUNCTION values(Vals const &...args) noexcept : m_vals(args...) {}
+
+            constexpr GT_FUNCTION values(tuple<Vals...> &&args) noexcept : m_vals(wstd::move(args)) {}
+            constexpr GT_FUNCTION values(tuple<Vals...> const &args) noexcept : m_vals(args) {}
+
+            values() = default;
+            values(values const &) = default;
+            values(values &&) = default;
+            values &operator=(values const &) = default;
+            values &operator=(values &&) = default;
+
+            GT_TUPLE_UTIL_FORWARD_GETTER_TO_MEMBER(values, m_vals);
+
+            friend keys hymap_get_keys(values const &) { return {}; }
+        };
+
+        template <>
+        template <>
+        struct keys<>::values<> {
+            friend values tuple_getter(values const &) { return {}; }
```

Comment:
Is this a workaround for a certain compiler. Could you add some comments what the different workarounds are about, if you remember.

Reply:
No, it is not a workaround.
The default template definition is ambiguous in the empty case because element wise ctors and default ctor.